llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -978,6 +978,21 @@ static void simplifyRecipe(VPRecipeBase &R, VPTypeAnalysis &TypeInfo) {
       TypeInfo.inferScalarType(R.getOperand(1)) ==
           TypeInfo.inferScalarType(R.getVPSingleValue()))
     return R.getVPSingleValue()->replaceAllUsesWith(R.getOperand(1));
+
+  if (match(&R, m_Not(m_VPValue(A))) && isa<VPWidenRecipe>(A) &&
+      A->getNumUsers() == 1) {
```

Comment:
The transformation modifies `A` in-place, hence checking that it has a single user (interested in this change). However, the intention is to collect `R` - worth also check if it has but a single user?
A comment could be added explaining this.

Reply:
Added, thanks